src/account/UpgradeableModularAccount.sol

```diff
@@ -155,6 +158,36 @@ contract UpgradeableModularAccount is
         return execReturnData;
     }
 
+    /// @notice Execution function that allows UO context to be passed to execution hooks
+    /// @dev This function is only callable by the EntryPoint
+    function executeUserOp(PackedUserOperation calldata userOp, bytes32) external {
```

Comment:
Agree with Adam. 

> Or do we have non-validation-associated exec hooks that also require UO context?

Looks like that is the case from the code.

But in this flow, both exec hooks (need or not UO context) are running twice.

For normal exec hooks need UO context, would it run not in this flow and revert the call. See comment below.

Reply:
It is currently both pre- and post hooks firing twice when we need UO context, right? 

If we need to have non-validation-associated exec hooks that also require UO context: Couldn't we also add a mechanism to figure out in the fallback function/wrapNativeFunction() modifier` if the self-call we are making originated from `executeUserOp()`, because in that case we are already “taking care” of the execution hooks? Perhaps a transient storage flag?

Reply:
Good find! 

The flag solution sounds good, will think through security if multiple transactions are happening in parallel/account is reentered

Another possible solution I'm thinking of is to force all exec hooks to not use UO context (or validation context), and all permission hooks to use UO context